File: ProductManagement/Client-Side/src/app/viewModels/product-management-viewModels/product-management-viewModels/response/response-get-products-viewModel.ts
export class ResponseGetProductsViewModel {
    constructor(data, totalCount) {
      this.data = data;
      this.totalCount = totalCount;
    }

    public data: Array<ResponseGetProductsViewModelItem>;
    public totalCount: number;
}

export class ResponseGetProductsViewModelItem {
    constructor(id, name, categoryId, categoryType) {
      this.id = id;
      this.name = name;
      this.categoryId = categoryId;
      this.categoryType = categoryType;
    }
  
    public id: number;
    public name: string;
    public categoryId: number;
    public categoryType: string;
}

File: ProductManagement.ViewModels/ProductManagementViewModels/Response/ResponseGetProductsViewModel.cs
using System.Collections.Generic;

namespace ProductManagement.ViewModels.ProductManagementViewModels.Response
{
    public class ResponseGetProductsViewModel
    {
        public List<ResponseGetProductsViewModelItem> Data { get; set; }
        public int TotalCount { get; set; }

        public ResponseGetProductsViewModel()
        {
            Data = new List<ResponseGetProductsViewModelItem>();
        }
    }

    public class ResponseGetProductsViewModelItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int CategoryId { get; set; }
        public string CategoryType { get; set; }
    }
}
